WalletWasabi.Backend/Controllers/BlockchainController.cs

```diff
@@ -416,4 +416,88 @@ private async Task<StatusResponse> FetchStatusAsync(CancellationToken cancellati
 
 		return status;
 	}
+
+	[HttpGet("unconfirmed-transaction-chain")]
+	[ProducesResponseType(200)]
+	[ProducesResponseType(400)]
+	public async Task<List<UnconfirmedTransactionChainItem>> GetUnconfirmedTransactionChainAsync([FromQuery, Required] string transactionId, CancellationToken cancellationToken)
+	{
+		uint256 txId = new(transactionId);
+
+		var cacheKey = $"{nameof(GetUnconfirmedTransactionChainAsync)}_{txId}";
+		var cacheOptions = new MemoryCacheEntryOptions { AbsoluteExpirationRelativeToNow = TimeSpan.FromMinutes(3) };
+
+		return await Cache.GetCachedResponseAsync(
+			cacheKey,
+			action: (string request, CancellationToken token) => GetUnconfirmedTransactionChainNoCacheAsync(txId, token),
+			options: cacheOptions,
+			cancellationToken);
+	}
+
+	private async Task<List<UnconfirmedTransactionChainItem>> GetUnconfirmedTransactionChainNoCacheAsync(uint256 txId, CancellationToken cancellationToken)
+	{
+		Dictionary<uint256, Transaction> transactionsLocalCache = new();
+		Dictionary<uint256, UnconfirmedTransactionChainItem> unconfirmedTxsChainById = new();
+		var mempool = Global.HostedServices.Get<MempoolMirror>();
```

Comment:
Make MempoolMirror public in Global? Purpose: spare the time of getting the service everytime.